pkg/kubelet/dockershim/network/kubenet/kubenet_linux.go

```diff
@@ -87,38 +114,43 @@ type kubenetNetworkPlugin struct {
 	hostportSyncer  hostport.HostportSyncer
 	hostportManager hostport.HostPortManager
 	iptables        utiliptables.Interface
+	iptablesv6      utiliptables.Interface
 	sysctl          utilsysctl.Interface
 	ebtables        utilebtables.Interface
 	// binDirs is passed by kubelet cni-bin-dir parameter.
 	// kubenet will search for CNI binaries in DefaultCNIDir first, then continue to binDirs.
-	binDirs           []string
-	nonMasqueradeCIDR string
-	podCidr           string
-	gateway           net.IP
+	binDirs            []string
+	nonMasqueradeCIDRs []string
+	podCIDRs           []*net.IPNet
```

Comment:
@khenidak I think we need locking around podCIDRs too, much like with podIPs. It appears this gets accessed from both Events and a bunch of other places and these may well be from different goroutines. Since you've pushed mu.Lock()/Unlock() more granular, it's no longer protecting podCIDRs anymore.

Reply:
PodCIDRs only written to in `Event()`, concurrently read in`SetupPod()` and other places, However all the other places are  preceded by `Status()` call which does not return true until PodCIDR  and CNI config has been set in `Event()`. so r/w does not happen concurrently.